File: src/main/java/com/uiptv/model/Category.java
package com.uiptv.model;

import com.uiptv.shared.BaseJson;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.json.JSONObject;

import static com.uiptv.util.StringUtils.safeGetString;

@Data
@NoArgsConstructor
public class Category extends BaseJson {
    private String dbId, accountId, accountType, categoryId, title, alias;
    private String extraJson;
    private boolean activeSub;
    private int censored;

    public Category(String categoryId, String title, String alias, boolean activeSub, int censored) {
        this.categoryId = categoryId;
        this.title = title;
        this.alias = alias;
        this.activeSub = activeSub;
        this.censored = censored;
    }

    public static Category fromJson(String json) {
        try {
            JSONObject jsonObj = new JSONObject(json);
            Category category = new Category();
            category.setDbId(safeGetString(jsonObj, "dbId"));
            category.setAccountId(safeGetString(jsonObj, "accountId"));
            category.setAccountType(safeGetString(jsonObj, "accountType"));
            category.setCategoryId(safeGetString(jsonObj, "categoryId"));
            category.setTitle(safeGetString(jsonObj, "title"));
            category.setAlias(safeGetString(jsonObj, "alias"));
            category.setExtraJson(safeGetString(jsonObj, "extraJson"));
            category.setActiveSub(jsonObj.optBoolean("activeSub"));
            category.setCensored(jsonObj.optInt("censored"));
            return category;
        } catch (Exception e) {
            return null;
        }
    }
}

File: server/src/main/kotlin/com/uiptv/model/Category.kt
package com.uiptv.model

import com.uiptv.shared.BaseJson
import com.uiptv.util.StringUtils.safeGetString
import com.uiptv.util.json.KJsonObject

data class Category @JvmOverloads constructor(
    var dbId: String? = null,
    var accountId: String? = null,
    var accountType: String? = null,
    var categoryId: String? = null,
    var title: String? = null,
    var alias: String? = null,
    var extraJson: String? = null,
    @get:JvmName("isActiveSub")
    var activeSub: Boolean = false,
    var censored: Int = 0
) : BaseJson() {
    constructor(
        categoryId: String?,
        title: String?,
        alias: String?,
        activeSub: Boolean,
        censored: Int
    ) : this(null, null, null, categoryId, title, alias, null, activeSub, censored)

    companion object {
        @JvmStatic
        fun fromJson(json: String): Category? {
            return try {
                val jsonObj = KJsonObject(json)
                Category(
                    dbId = safeGetString(jsonObj, "dbId"),
                    accountId = safeGetString(jsonObj, "accountId"),
                    accountType = safeGetString(jsonObj, "accountType"),
                    categoryId = safeGetString(jsonObj, "categoryId"),
                    title = safeGetString(jsonObj, "title"),
                    alias = safeGetString(jsonObj, "alias"),
                    extraJson = safeGetString(jsonObj, "extraJson"),
                    activeSub = jsonObj.optBoolean("activeSub"),
                    censored = jsonObj.optInt("censored")
                )
            } catch (_: Exception) {
                null
            }
        }
    }
}
